src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatisticsModel.java

```diff
@@ -0,0 +1,103 @@
+package de.dennisguse.opentracks.viewmodels;
+
+import android.app.Application;
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import java.util.List;
+
+import de.dennisguse.opentracks.content.data.TrackPoint;
+import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
+import de.dennisguse.opentracks.util.PreferencesUtils;
+import de.dennisguse.opentracks.util.UnitConversions;
+
+public class IntervalStatisticsModel extends AndroidViewModel {
+
+    private MutableLiveData<IntervalStatistics> intervalStats;
+    private IntervalOption interval;
+
+    public IntervalStatisticsModel(@NonNull Application application) {
+        super(application);
+        interval = IntervalOption.OPTION_1;
+    }
+
+    public LiveData<IntervalStatistics> getIntervalStats(long trackId, IntervalOption interval) {
+        if (intervalStats == null || this.interval != interval) {
+            intervalStats = new MutableLiveData<>();
+            this.interval = interval;
+            loadIntervalStats(trackId);
+        }
+        return intervalStats;
+    }
+
+    public LiveData<IntervalStatistics> getIntervalStats(long trackId) {
+        return getIntervalStats(trackId, interval);
+    }
+
+    public void invalidate() {
+        intervalStats = null;
+    }
+
+    private void loadIntervalStats(long trackId) {
+        new Thread(() -> {
+            Context context = getApplication().getApplicationContext();
+            ContentProviderUtils contentProviderUtils = new ContentProviderUtils(context);
+            List<TrackPoint> trackPointList = contentProviderUtils.getTrackPoints(trackId);
+
+            IntervalStatistics intervalStatistics = new IntervalStatistics();
+            float distanceInterval = PreferencesUtils.isMetricUnits(context) ? (float) (interval.getValue() * UnitConversions.KM_TO_M) : (float) (interval.getValue() * UnitConversions.MI_TO_M);
+            intervalStatistics.build(trackPointList, distanceInterval);
+
+            intervalStats.postValue(intervalStatistics);
+        }).start();
+    }
+
+    public enum IntervalOption {
+        OPTION_1(1),
```

Comment:
i guess this will be either 1mile / 1km, right?
So, we are independent of the actual unit

Reply:
Yes, it is.